File: src/GCov/State/ShowNonProjectSourcesState.java
package GCov.State;

import com.intellij.openapi.components.PersistentStateComponent;
import com.intellij.openapi.components.ServiceManager;
import com.intellij.openapi.components.State;
import com.intellij.openapi.project.Project;
import com.intellij.util.xmlb.XmlSerializerUtil;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@State(name = "GCoverageShowNonProjectSources")
public class ShowNonProjectSourcesState implements PersistentStateComponent<ShowNonProjectSourcesState> {

    public boolean showNonProjectSources = false;

    public static ShowNonProjectSourcesState getInstance(@NotNull Project project) {
        return ServiceManager.getService(project,ShowNonProjectSourcesState.class);
    }

    @Nullable
    @Override
    public ShowNonProjectSourcesState getState() {
        return this;
    }

    @Override
    public void loadState(@NotNull ShowNonProjectSourcesState showNonProjectSourcesState) {
        XmlSerializerUtil.copyBean(showNonProjectSourcesState,this);
    }

}

File: src/gcov/state/ShowNonProjectSourcesState.kt
package gcov.state

import com.intellij.openapi.components.PersistentStateComponent
import com.intellij.openapi.components.ServiceManager
import com.intellij.openapi.components.State
import com.intellij.openapi.project.Project
import com.intellij.util.xmlb.XmlSerializerUtil

@State(name = "GCoverageShowNonProjectSources")
class ShowNonProjectSourcesState : PersistentStateComponent<ShowNonProjectSourcesState> {

    var showNonProjectSources = false

    override fun getState(): ShowNonProjectSourcesState = this

    override fun loadState(showNonProjectSourcesState: ShowNonProjectSourcesState) = XmlSerializerUtil.copyBean(showNonProjectSourcesState, this)

    companion object {

        fun getInstance(project: Project): ShowNonProjectSourcesState = ServiceManager.getService(project, ShowNonProjectSourcesState::class.java)!!
    }

}
